dynawo/sources/Solvers/SolverSIM/DYNSolverSIM.cpp

```diff
@@ -293,162 +280,61 @@ SolverSIM::calculateIC() {
-          }
-        break;
-      }
+    // Call the algebraic solver, analyze the result and detect mode or z change if the algebraic solver has converged
+    SolverStatus_t status;
+    if (model_->sizeY() != 0) {
+      int flag = callAlgebraicSolver();
+      status = analyzeResult(flag);
+      if (status != NON_CONV)
+        updateZAndMode(status);
+    } else {
+      updateZAndMode(status);
     }
-  } while (redoStep);
-
-  // Limitation to end up the simulation at tEnd
-  updateTimeStep(tNxt);
-  ++stats_.nst_;
-}
-
-void SolverSIM::solveWithoutStepRecalculation(double& tNxt) {
-  int counter = 0;
-  bool redoStep = false;
-  saveContinuousVariables();
-
-  do {
-    // If we have more than maxNewtonTry consecutive divergence or root changes, the simulation is stopped
-    handleMaximumTries(counter);
-
-    // New time step value
-    h_ = hNew_;
-
-    // Call the Newton-Raphson solver and analyze the root evolution
-    SolverStatus_t status = solve();
-    skipAlgebraicResidualsEvaluation_ = false;
 
     switch (status) {
-      /* NON_CONV: the Newton-Raphson algorithm fails to converge
+      /* NON_CONV: the algebraic solver fails to converge
        *   => The time step is decreased
-       *   => The LU decomposition is forced for the next time steps
-       *   => The initial point for the next time step (h = h/2) is the initial point from the previous accepted time step
+       *   => The LU decomposition is forced for the next time step.
+       *   => The initial point for the next time step (h = h * kReduceStep) is the initial point from the previous accepted time step
       */
       case NON_CONV: {
         handleDivergence(redoStep);
-        restoreContinuousVariables();
         break;
       }
       /*
-       * CONV: the Newton-Rapshon algorithm converges and there is no discrete value or mode change
-       * => The time step is increased (if possible)
-       * => The LU decomposition is avoided for the next time steps (we have come back to a stabilized situation)
+       * CONV: the algebraic solver converges and there is no discrete value or mode change
+       * => The time step is increased (h = h / kReduceStep)
+       * => The LU decomposition is avoided for the next time step.
+       * => The initial point for the next time step is the final point of this time step
       */
       case CONV: {
         handleConvergence(redoStep);
-        skipAlgebraicResidualsEvaluation_ = optimizeAlgebraicResidualsEvaluations_;
-        break;
-      }
-      /*
-       * ROOT_ALG: algebraic change
-       * => a restoration of the algebraic variables will be called by the simulation
-       * => The current time step is accepted.
-       * => The next time step will be done with the current time step (neither increase nor decrease)
-      */
-      case ROOT_ALG: {
-        handleAlgebraicRoot(redoStep);
-        skipAlgebraicResidualsEvaluation_ = optimizeAlgebraicResidualsEvaluations_;
```

Comment:
In the case of ROOT_ALG we were allowing to skip the next evalF, I am not sure it is still true after the change?

Reply:
It has been moved to the reinit() method and is then set only when the reinit() is done, which wasn't the case before.

If I remember correctly, it was the original unconsistency that motivated the dev because it should be skip if and only if the reinit() is called so it is also depending on the minimumModeChangeTypeForAlgebraicRestoration_ value. Otherwise if no reinit() and algebraic change, then the algebraic equations are not correct (at least one)